### DISCUSSION TITLE
[错误报告]: 资源名称识别错误

### DISCUSSION THREAD

**Discussion by panqx99** [jxxghp/MoviePilot#787] [讨论]
### 确认

- [X] 我的版本是最新版本，我的版本号与 [version](https://github.com/jxxghp/MoviePilot/releases/latest) 相同。
- [X] 我已经 [issue](https://github.com/jxxghp/MoviePilot/issues) 中搜索过，确认我的问题没有被提出过。
- [X] 我已经 [Telegram频道](https://t.me/moviepilot_channel) 中搜索过，确认我的问题没有被提出过。
- [X] 我已经修改标题，将标题中的 描述 替换为我遇到的问题。

### 当前程序版本

1.2.8

### 问题类型

主程序运行问题

### 问题描述

影视资源下载后往媒体库转移时，名称识别不正确。如：资源名为以下名称时，通过moviepilot识别成了“爱的诗韵”，同样的名称在nastool可以正常识别为“寻梦环游记”。希望大佬可以优化一下命名识别规则，多谢~

[寻梦环游记].Coco.2017.BluRay.1080p.x264.DTS.4Audios-CMCT

**Comment by cng10001** [comment]
感觉还是标题里的标点符号影响了

> []

**Comment by panqx99** [comment]
我也多试了几遍，发现去掉2017就识别正确了，好像不是符号的问题
[寻梦环游记].Coco.2017.BluRay.1080p.x264.DTS.4Audios-CMCT　⇒　"爱的诗韵(2017)"
[寻梦环游记].Coco.BluRay.1080p.x264.DTS.4Audios-CMCT　⇒　"寻梦环游记(2017)"

**Comment by c77187326** [comment]
感觉是符号问题，我上次也是，去掉符号[  ]后就正常了

**Comment by jxxghp** [comment]
![image](https://github.com/jxxghp/MoviePilot/assets/51039935/2f438802-0331-4116-a7d4-3534745c9424)

这个名字识别是正常的，是种子命名的原因吧。

**Comment by cng10001** [comment]
> 我也多试了几遍，发现去掉2017就识别正确了，好像不是符号的问题 [寻梦环游记].Coco.2017.BluRay.1080p.x264.DTS.4Audios-CMCT　⇒　"爱的诗韵(2017)" [寻梦环游记].Coco.BluRay.1080p.x264.DTS.4Audios-CMCT　⇒　"寻梦环游记(2017)"

 [#因为我把符号去掉，电影年份保留也能识别，哈哈哈]
![Screenshot_2023-10-07-17-31-27-689_com android browser-04](https://github.com/jxxghp/MoviePilot/assets/115156545/acaee448-23c7-4596-bb04-4f12b6b53bcb)

**Comment by panqx99** [comment]
[]去掉后果然能正确识别，但是我看了一下现在好多种子都是这样命名，中文名用[]括起来，这种情况只能手动的去重新整理吗。
另外发现，moviepilot转移的时候是用影视资源的文件名去名称识别的，用这个种子的名称是能正常识别出来的。

种子名称：寻梦环游记.2017.1080p.国粤台英四语.中英字幕￡CMCT梦幻　⇒　名称识别测试正常
影视资源文件名：[寻梦环游记].Coco.2017.BluRay.1080p.x264.DTS.4Audios-CMCT.mkv　⇒　名称识别测试错误

**Comment by AisukaYuki** [comment]
通用规则，把文件名开头是[中文标题].EngTitle.2000.xxxxxxxx.mkv 这样的形式的命名，通通去掉[]
`^\[(.*[\u4e00-\u9fa5])\] => \1`
![ed165e0c-bb58-4a86-b6e1-44b32890027f](https://github.com/jxxghp/MoviePilot/assets/17586327/a3ff3440-b796-494f-92fd-233a32d0bff7)
![090d16d0-7a44-4a5a-8608-df63938eafc5](https://github.com/jxxghp/MoviePilot/assets/17586327/7d69dd0c-14d6-4bac-a4fd-f950ea816802)

**Reply by cng10001** [reply]
研究了下，如果说因为**开头[]中是中文**的问题，那要求是改善**开头[接中文字**的情况，我觉得应该是`^开头\[中括号[\u4e00-\u9fa5]中文字+一次或多次\]中括号.*?其他可能的字符
`
如果`.*在前面[\u4e00-\u9fa5]中文字`，就有可能是**先接英文再是中文**的情况比如`[X战警]`

**Comment by panqx99** [comment]
> 通用规则，把文件名开头是[中文标题].EngTitle.2000.xxxxxxxx.mkv 这样的形式的命名，通通去掉[] `^\[(.*[\u4e00-\u9fa5])\] => \1` ![ed165e0c-bb58-4a86-b6e1-44b32890027f](https://user-images.githubusercontent.com/17586327/273545717-a3ff3440-b796-494f-92fd-233a32d0bff7.png) ![090d16d0-7a44-4a5a-8608-df63938eafc5](https://user-images.githubusercontent.com/17586327/273546316-7d69dd0c-14d6-4bac-a4fd-f950ea816802.png)

完美解决方案，多谢大佬！！
